src/Module/Objects.php

```diff
@@ -42,7 +42,14 @@ public static function rawContent(array $parameters = [])
 			}
 		}
 
-		$data = ActivityPub\Transmitter::createObjectFromItemID($item['id']);
+		$activity = ActivityPub\Transmitter::createActivityFromItem($item['id'], true);
+		// Only display "Create" activity objects here, no reshares or anything else
+		if (!is_array($activity['object']) || ($activity['type'] != 'Create')) {
+			throw new \Friendica\Network\HTTPException\NotFoundException();
```

Comment:
Why not redirect to the target post that was reshared?

Reply:
Sounds like some idea.

Reply:
I thought about it. This would be wrong. System would want to fetch this URL, but will be redirected to another one. This could confuse systems and wouldn't serve any purpose. It would be an idea if this was some front end endpoint - but this is purely a machine endpoint.